### PR TITLE
Lecture 6 Hometask

### PR DESCRIPTION
**Pull request by carmenhorendum**

### CODE REVIEW

File: Lecture5/src/test/java/HttpClientRequestExecutor.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	private static final String USER_AGENT = "Mozilla/5.0 (Windows NT 6.1; WOW64) " +
+			"AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2785.143 Safari/537.36";
+	private static HttpClient client = HttpClientBuilder.create().build();
+
+	@Override
+	public String makeRequest(String url, Method method, Map<String, String> params)
+			throws IOException, URISyntaxException {
+		switch (method) {
+			case GET:
+				for (Map.Entry<String, String> entry : params.entrySet()) {
```
Comment (nemec8g) [inline Lecture5/src/test/java/HttpClientRequestExecutor.java:26]: move common code in private method and reuse

File: Lecture5/src/test/java/HttpClientRequestExecutor.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+					StringBuffer bodyGet = new StringBuffer();
+					String line;
+					while ((line = rd.readLine()) != null) {
+						bodyGet.append(line);
+					}
+					return String.valueOf(bodyGet);
+				}
+				break;
+			case POST:
+				return "";
```
Comment (nemec8g) [inline Lecture5/src/test/java/HttpClientRequestExecutor.java:47]: throw new not implemented exception in all not implemented cases

File: Lecture5/src/test/java/HttpClientRequestExecutor.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+			case PUT:
+				for (Map.Entry<String, String> entry : params.entrySet()) {
+					URI uriPut = new URIBuilder()
+							.setScheme(PROTOCOL)
+							.setHost(url)
+//                            .setPath(ENDPOINT)
```
Comment (nemec8g) [inline Lecture5/src/test/java/HttpClientRequestExecutor.java:53]: remove commented code

File: Lecture5/src/test/java/RequestExecutor.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+public interface RequestExecutor {
+	String makeRequest(String url, Method method, Map<String, String> params) throws URISyntaxException, IOException;
```
Comment (nemec8g) [inline Lecture5/src/test/java/RequestExecutor.java:6]: catch those exception in `Executors` and throw runtime exception if you need

File: Lecture5/src/test/java/Test.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.Map;
+
+public class Test {
+	private static final String BASE_URI = "api.github.com/repos/carmenhorendum/AutomationSchool";
+	private static final String ENDPOINT = "/collaborators";
+	private static final String ENDPOINT_2 = "/collaborators/nemec8g";
+	private static final String OAUTH_TOKEN = "";
+
+	public static void main(String[] args) throws URISyntaxException, IOException {
+		RequestExecutor httpClientRequestExecutor = new HttpClientRequestExecutor();
```
Comment (nemec8g) [inline Lecture5/src/test/java/Test.java:17]: move common code in private method with param (RequestExecutor) and reuse

File: Lecture6/src/test/java/PositiveGrid.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+		element.sendKeys("");
+		element.submit();
+		assertEquals(driver.getTitle(), "Positive Grid");
+	}
+
+	@Test(description = "Navigates to the store page", dependsOnMethods = "loginAsAdmin")
+	public void navigateStore() {
+		WebElement element = driver.findElement(By.xpath("//*[@id=\"menu\"]/div[1]/a[2]"));
+//		WebElement element = driver.findElement(By.className("nav-item-text"));
+		try {
```
Comment (nemec8g) [inline Lecture6/src/test/java/PositiveGrid.java]: use  FluentWait  to wait some condition

File: Lecture6/src/test/java/PositiveGrid.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+		element.sendKeys ("");
+		element = driver.findElement(By.name("password"));
+		element.sendKeys("");
+		element.submit();
+		assertEquals(driver.getTitle(), "Positive Grid");
+	}
+
+	@Test(description = "Navigates to the store page", dependsOnMethods = "loginAsAdmin")
+	public void navigateStore() {
+		WebElement element = driver.findElement(By.xpath("//*[@id=\"menu\"]/div[1]/a[2]"));
```
Comment (nemec8g) [inline Lecture6/src/test/java/PositiveGrid.java]: bad locator - ordering may change

File: Lecture6/src/test/java/PositiveGrid.java [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+			e.printStackTrace();
+		}
+		element.click();
+//		driver.get("https://www.positivegrid.com/plugins/");
+		assertEquals(driver.getTitle(), "Online Store - Desktop Plugins Software - Positive Grid");
+	}
+
+	@Test(description = "Adds product to cart", dependsOnMethods = "navigateStore")
+	public void addProductToCart() {
+		WebElement element = driver.findElement(By.cssSelector("#product-action-btn-container > input:nth-child(6)"));
```
Comment (nemec8g) [inline Lecture6/src/test/java/PositiveGrid.java]: bad locator ordering childs may change

File: Lecture6/src/test/java/PositiveGrid.java [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+		WebElement element = driver.findElement(By.className("nav-item-icon"));
+		element.click();
+		element = driver.findElement(By.className("data-login-PG-btn"));
+		element.click();
+		assertEquals(driver.getTitle(), "Positive Grid");
+	}
+
+	@Test(description = "Enters valid login data", dependsOnMethods = "openSignInPage")
+	public void loginAsAdmin() {
+		WebElement element = driver.findElement(By.id("inputEmail"));
```
Comment (nemec8g) [inline Lecture6/src/test/java/PositiveGrid.java]: please move all locators to separate class with constants

File: Lecture6/src/test/java/BaseTest.java [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import org.testng.ITestContext;
+import org.testng.annotations.BeforeSuite;
+
+public class BaseTest {
+	@BeforeSuite(alwaysRun = true)
+	public void setupBeforeSuite(ITestContext context) {
+//		String seleniumHost = context.getCurrentXmlTest().getParameter("selenium.host");
```
Comment (nemec8g) [inline Lecture6/src/test/java/BaseTest.java]: do not leave commented code

File: Lecture6/src/test/java/PositiveGrid.java (hunk 7 above)
Comment (carmenhorendum) [reply Lecture6/src/test/java/PositiveGrid.java]: fixed

File: Lecture6/src/test/java/PositiveGrid.java (hunk 8 above)
Comment (carmenhorendum) [reply Lecture6/src/test/java/PositiveGrid.java]: fixed

File: Lecture6/src/test/java/PositiveGrid.java (hunk 9 above)
Comment (carmenhorendum) [reply Lecture6/src/test/java/PositiveGrid.java]: done

File: Lecture6/src/test/java/BaseTest.java (hunk 10 above)
Comment (carmenhorendum) [reply Lecture6/src/test/java/BaseTest.java]: removed

File: Lecture6/src/test/java/PositiveGrid.java (hunk 6 above)
Comment (carmenhorendum) [reply Lecture6/src/test/java/PositiveGrid.java]: WebDriverWait used

File: Lecture5/src/test/java/HttpClientRequestExecutor.java (hunk 3 above)
Comment (carmenhorendum) [reply Lecture5/src/test/java/HttpClientRequestExecutor.java:53]: done

File: Lecture5/src/test/java/RequestExecutor.java (hunk 4 above)
Comment (carmenhorendum) [reply Lecture5/src/test/java/RequestExecutor.java:6]: done

File: Lecture5/src/test/java/HttpClientRequestExecutor.java (hunk 2 above)
Comment (carmenhorendum) [reply Lecture5/src/test/java/HttpClientRequestExecutor.java:47]: done

**Pull request merged**